bundles/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/handler/FridgeFreezerChannelSelector.java

```diff
@@ -160,6 +161,15 @@ public State getState(String s) {
         return null;
     }
 
+    public State getTemperatureState(String s) {
+        try {
+            return ExtendedDeviceStateUtil.getTemperatureState(s);
+        } catch (Exception e) {
+            logger.error("An exception occurred while converting '{}' into a State", s);
+            return null;
```

Comment:
Rather than null, wouldn't it be better to return UndefType.UNDEF ?
In case there is a valid reason to return null, please add `@Nullable` before State in the method declaration.

Reply:
Good idea, fixed.